cashctrl_ledger/tests/test_ledger.py

```diff
@@ -1,124 +1,171 @@
-"""Unit tests for ledger accessors, mutators and mirroring."""
+"""Unit tests for ledger accessor, mutator, and mirror methods."""
 
 import pytest
 import pandas as pd
-from requests.exceptions import RequestException
-from pyledger.tests import BaseTestLedger
+from requests import RequestException
 # flake8: noqa: F401
-from base_test import initial_ledger
+from base_test import initial_engine
+from pyledger.tests import BaseTestLedger
 from io import StringIO
 
 
-LEDGER_IDS = set(
-    BaseTestLedger.LEDGER_ENTRIES["id"].astype(str).unique()
-).difference(['15', '16', '17', '18'])
-
-
 class TestLedger(BaseTestLedger):
-    @pytest.fixture(scope="class")
-    def ledger(self, initial_ledger):
-        initial_ledger.restore(
-            accounts=self.ACCOUNTS, tax_codes=self.TAX_CODES, settings=self.SETTINGS
-        )
-        return initial_ledger
-
-    @pytest.mark.parametrize("ledger_id", LEDGER_IDS)
-    def test_add_ledger_entry(self, ledger, ledger_id):
-        super().test_add_ledger_entry(ledger, ledger_id)
-
-    def test_add_ledger_with_non_existing_tax(self, ledger):
-        # Adding a ledger entry with non existing tax code should raise an error
-        target = self.LEDGER_ENTRIES.query("id == '1'").copy()
-        target.loc[0, "tax_code"] = "Test_Non_Existent_TAX_code"
+    ACCOUNTS = BaseTestLedger.ACCOUNTS.copy()
+    # Set the default root node, as CashCtrl does not allow the creation of root nodes
+    ACCOUNTS.loc[:, "group"] = "/Assets"
+    # TODO: Remove when Assets will be implemented
+    ACCOUNTS.loc[ACCOUNTS["currency"] == "JPY", "currency"] = "USD"
+
+    TAX_CODES = BaseTestLedger.TAX_CODES.copy()
+    # Assign a default account to TAX_CODES where account is missing,
+    # CashCtrl does not support tax codes without accounts assigned
+    default_account = TAX_CODES.query("id == 'IN_STD'")["account"].values[0]
+    TAX_CODES.loc[TAX_CODES["account"].isna(), "account"] = default_account
```

Comment:
I believe this codes is duplicated. 
Let's find a way to abstract it or get rid of it.

Feel free to declare as out of scope for this PR and open an issue to keep track of the comment.

Reply:
Here is the issue for this: https://github.com/macxred/cashctrl_ledger/issues/89